Adds TSC_COMPILE_ON_ERROR env var... (#6931)
facebook/create-react-app: docusaurus/docs/adding-typescript.md

Next, rename any file to be a TypeScript file (e.g. `src/index.js` to `src/index.tsx`) and **restart your development server**!

> Type errors will show up in the same console as the build one. By default, Create React App prevents you from running the dev server if your code has type errors. If you introduce type errors to your project, you have to fix or ignore them before you continue development.

Author: Hey @mrmckeb 

I've changed the above to the following:

>Type errors will show up in the same console as the build one. You'll have to fix these type errors before you continue development or build your project. For advanced configuration, [see here](advanced-configuration.md).

Now it doesn't explicitly **or even implicitly** encourage anyone to do this =)

Thanks for your feedback!

> You can remove this restriction by running your app with the `TSC_COMPILE_ON_ERROR` environment variable set to `true`, for example, by adding `TSC_COMPILE_ON_ERROR=true` to your `.env` file. [Read more about setting environment variables here](https://facebook.github.io/create-react-app/docs/adding-custom-environment-variables).

Reviewer: ```suggestion
You can remove this restriction by running your app with the `TSC_COMPILE_ON_ERROR` environment variable set to `true`, for example, by adding `TSC_COMPILE_ON_ERROR=true` as documented in our [advanced configuration](advanced-configuration.md).
```

> 

Reviewer: Can you add this new variable to `advanced-configuration.md`?

Author: Hey @ianschmitz , just added this, sorry I forgot to do it the first time around.

Next, rename any file to be a TypeScript file (e.g. `src/index.js` to `src/index.tsx`) and **restart your development server**!

Type errors will show up in the same console as the build one. By default, Create React App prevents you from running the dev server if your code has type errors. If you introduce type errors to your project, you have to fix or ignore them before you continue development.

> You can remove this restriction by running your app with the `TSC_COMPILE_ON_ERROR` environment variable set to `true`, for example, by adding `TSC_COMPILE_ON_ERROR=true` as documented in our [advanced configuration](advanced-configuration.md).

Reviewer: Good point. I agree this is likely an advanced scenario for those porting legacy code. New code bases really shouldn't be using this option.
